src/Storages/VirtualColumnsDescription.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <Storages/ColumnsDescription.h>
+#include <Common/MultiVersion.h>
+
+namespace DB
+{
+
+struct VirtualColumnDescription : public ColumnDescription
+{
+public:
+    using Self = VirtualColumnDescription;
+    VirtualsKind kind;
+
+    VirtualColumnDescription() = default;
+    VirtualColumnDescription(String name_, DataTypePtr type_, ASTPtr codec_, String comment_, VirtualsKind kind_);
+
+    bool isEphemeral() const { return kind == VirtualsKind::Ephemeral; }
+    bool isPersistent() const { return kind == VirtualsKind::Persistent; }
+    const String & getName() const { return name; }
```

Comment:
nit: maybe a comment that this method is required for the `boost::multi_index_container`